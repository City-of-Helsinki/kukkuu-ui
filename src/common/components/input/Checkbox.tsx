import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { FieldInputProps } from 'formik';

import styles from './input.module.scss';

type CheckboxProps = {
  className?: string;
  label: string;
  required?: boolean;
  name: string;
};
// eslint-disable-next-line @typescript-eslint/no-explicit-any
const Checkbox: FunctionComponent<CheckboxProps & FieldInputProps<any>> = ({
  className,
  label,
  required,
  name,
  ...rest
}) => {
  return (
    <div
      className={classnames(
        styles.inputWrapper,
        styles.checkboxInput,
        className
      )}
    >
      <div className={styles.inputTypeWrapper}>
        <input type="checkbox" id={name} {...rest} />
        {label && (
          <label htmlFor={name}>{required ? `${label}*` : label}</label>
        )}
      </div>
    </div>
  );
};

export default Checkbox;
