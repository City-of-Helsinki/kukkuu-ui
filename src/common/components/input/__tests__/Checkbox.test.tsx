import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Checkbox from '../Checkbox';

test('renders snapshot correctly', () => {
  const wrapper = shallow(
    <Checkbox
      name="thisCheckbox"
      label="bar"
      value={'wow'}
      onChange={jest.fn()}
      onBlur={jest.fn()}
    />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});

test('render label with asterisk if required props is true', () => {
  const wrapper = shallow(
    <Checkbox
      name="thisCheckbox"
      label="bar"
      value={'wow'}
      onChange={jest.fn()}
      onBlur={jest.fn()}
      required={true}
    />
  );
  expect(wrapper.find('label').text()).toContain('*');
});
