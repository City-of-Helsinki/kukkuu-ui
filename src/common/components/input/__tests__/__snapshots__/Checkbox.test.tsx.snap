// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<div
  className="inputWrapper checkboxInput"
>
  <div
    className="inputTypeWrapper"
  >
    <input
      id="thisCheckbox"
      onBlur={[MockFunction]}
      onChange={[MockFunction]}
      type="checkbox"
      value="wow"
    />
    <label
      htmlFor="thisCheckbox"
    >
      bar
    </label>
  </div>
</div>
`;
